frontend/lib/swr-cache/index.ts

```diff
@@ -72,18 +75,41 @@ export const withCache = <
    const logger: Logger = defaultLog;
 
    const requestRevalidation = async (variables: z.infer<VariablesSchema>) => {
-      fetch(`${APP_ORIGIN}/${endpoint}`, {
-         method: 'POST',
-         body: JSON.stringify(variables),
-      }).catch((error) => {
-         logger.error(
-            `${endpoint}.error: failed to trigger revalidation\n${printError(
-               error
-            )}`
-         );
+      const start = performance.now();
+      await new Promise<void>((resolve, reject) => {
+         // Don't delay the lambda beyond 50ms for revalidation
+         const timeout = setTimeout(() => {
+            const elapsed = performance.now() - start;
+            logger.warning.event(`${endpoint}.revalidation.timeout`);
+            logger.warning.timing(`${endpoint}.revalidation.timeout`, elapsed);
+            resolve();
+         }, 50);
+         const postData = JSON.stringify(variables);
+         const req = request({
+            hostname: VERCEL_URL || new URL(APP_ORIGIN).hostname,
+            ...(!VERCEL_URL && { port: new URL(APP_ORIGIN).port }),
+            path: `/${endpoint}`,
+            method: 'POST',
+         }).on('error', (error) => {
+            logger.error(
+               `${endpoint}.error: failed to trigger revalidation\n${printError(
+                  error
+               )}`
+            );
+            reject();
```

Comment:
Might want to record a stat here too... not sure how common it'll be, but it might be nice to know.